### PR TITLE
[flutter_tools] support multiple fuchsia devices

### LINKED ISSUE #55765: [fuchsia] flutter drive doesn't work with emulator devices
Using an emulator device (e.g. `fx emu -N`) and trying to use `flutter drive` to run a test on it, I get a device not found error (logs have been stripped down to the relevant parts):

```flutter drive -v -d step-atom-yard-juicy --target=test_driver/stock_view.dart
...
[  +25 ms] List of devices attached
[ +998 ms] 192.168.42.44 bless-fade-grill-slit
                    192.168.42.48 step-atom-yard-juicy
                    192.168.42.41 yard-frown-baton-ouch
[   +5 ms] executing: /usr/local/google/home/wrightgeorge/dev/flutter/bin/cache/artifacts/fuchsia/tools/device-finder resolve -device-limit 1 bless-fade-grill-slit
[  +15 ms] 192.168.42.44
...
[  +36 ms] No devices found with name or id matching 'step-atom-yard-juicy'
[   +9 ms] "flutter drive" took 1,345ms.
 
#0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
#1      DriveCommand.runCommand (package:flutter_tools/src/commands/drive.dart:145:7)
<asynchronous suspension>
#2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:760:18)
#3      _rootRunUnary (dart:async/zone.dart:1192:38)
#4      _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#5      _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
#6      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:686:45)
#7      Future._propagateToListeners (dart:async/future_impl.dart:715:32)
#8      Future._completeWithValue (dart:async/future_impl.dart:526:5)
#9      Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:560:7)
#10     _rootRun (dart:async/zone.dart:1184:13)
#11     _CustomZone.run (dart:async/zone.dart:1077:19)
#12     _CustomZone.runGuarded (dart:async/zone.dart:979:7)
#13     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1019:23)
#14     _microtaskLoop (dart:async/schedule_microtask.dart:43:21)
#15     _startMicrotaskLoop (dart:async/schedule_microtask.dart:52:5)
#16     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
#17     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)```

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Fixes https://github.com/flutter/flutter/issues/55765

We are currently only returning the first device from dev-finder, instead we need to look at the whole list.

### CODE REVIEW

File: packages/flutter_tools/lib/src/fuchsia/fuchsia_sdk.dart [hunk 1]
```diff
@@ -48,7 +48,7 @@ class FuchsiaSdk {
   /// Example output:
   ///    $ device-finder list -full
   ///    > 192.168.42.56 paper-pulp-bush-angel
-  Future<String> listDevices({ Duration timeout }) async {
+  Future<List<String>> listDevices({ Duration timeout }) async {
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/fuchsia/fuchsia_sdk.dart:51]: This change will require an update in google3.

File: packages/flutter_tools/test/general.shard/fuchsia/fuchsia_device_test.dart [hunk 2]
```diff
@@ -57,20 +57,19 @@ void main() {
```
Comment (zanderso) [inline packages/flutter_tools/test/general.shard/fuchsia/fuchsia_device_test.dart:57]: Is there already a test with multiple devices from `FuchsiaDevFinder.list()`?

File: packages/flutter_tools/lib/src/fuchsia/fuchsia_sdk.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/fuchsia/fuchsia_sdk.dart:51]: Crap :(

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/fuchsia/fuchsia_sdk.dart]: Updated API

File: packages/flutter_tools/test/general.shard/fuchsia/fuchsia_device_test.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/fuchsia/fuchsia_device_test.dart:79]: Test added

File: packages/flutter_tools/test/general.shard/fuchsia/fuchsia_device_test.dart [hunk 3]
```diff
@@ -1476,3 +1539,5 @@ class MockFuchsiaSdk extends Mock implements FuchsiaSdk {
   @override
   final FuchsiaDevFinder fuchsiaDevFinder;
 }
+
+class MockFuchsiaWorkflow extends Mock implements FuchsiaWorkflow {}
```
Comment (zanderso) [inline packages/flutter_tools/test/general.shard/fuchsia/fuchsia_device_test.dart:1543]: nit: missing newline at EOF

**Review by zanderso — APPROVED** [review]
lgtm

**Pull request merged**